src/citrine/resources/ara_job.py

```diff
@@ -0,0 +1,179 @@
+    failure_reason = properties.Optional(String(), "failure_reason")
+
+    def __init__(
+            self,
+            id: str,
+            task_type: str,
+            status: str,
+            dependencies: Set[str],
+            failure_reason: Optional[str]
+    ):
+        self.id = id
+        self.task_type = task_type
+        self.status = status
+        self.dependencies = dependencies
+        self.failure_reason = failure_reason
+
+
+class TaskNodeType(Property[TaskNode, dict]):
+    @property
+    def underlying_types(self):
+        return TaskNode
+
+    @property
+    def serialized_types(self):
+        return dict
+
+    def _serialize(self, value: TaskNode) -> dict:
+        result = {"id": value.id, "task_type": value.task_type, "status": value.status,
+                  "failure_reason": value.failure_reason, "dependencies": value.dependencies}
+        return result
+
+    def _deserialize(self, value: dict) -> TaskNode:
+        return TaskNode.build(value)
+
+
+class JobStatusResponse(Resource['JobStatusResponse']):
+    """
+    [ALPHA] a response to a job status check
+
+    Parameters
+    ----------
+    job_type: str
+        the type of job for this status report
+    status: str
+        the actual status of the job
+    tasks: List[TaskNode]
+        all of the constituent task required to complete this job
+    """
+
+    job_type = properties.String("job_type")
+    status = properties.String("status")
+    tasks = properties.List(TaskNodeType, "tasks")
+
+    def __init__(
+            self,
+            job_type: str,
+            status: str,
+            tasks: List[TaskNode]
+    ):
+        self.job_type = job_type
+        self.status = status
+        self.tasks = tasks
+
+
+class AraJobFramework(Resource["AraJobFramework"]):
+    """
+    [ALPHA] a resource for submitting a job and getting job status
+
+    Parameters
+    ----------
+    session: Session
+        the remote session for hitting framework endpoints
+    """
+    _path_template = 'projects/{project_id}'
+
+    def __init__(self, session: Session):
+        self.session = session
+
+    def build_ara_table(self, project: Project,
+                        ara_def: AraDefinition) -> Union[ApiError, AraJobSubmissionResponse]:
+        """
+        [ALPHA] submit an ara table construction job
+
+        This method makes a call out to the job framework to start a new job to build
+        the respective table for a given AraDefinition.
+
+        Parameters
+        ----------
+        project: Project
+            The project on behalf of which the job executes.
+        ara_def: AraDefinition
+            the ara definition describing the new table
+        """
+        job_id = uuid4()
+        url_suffix: str = "/ara-definitions{ara_definition}/versions/{version_number}/build?job_id={job_id}"
+        if ara_def.version_number is None:
+            return ApiError(400, "Version number was not present request", None)
```

Comment:
```suggestion
            return ApiError(400, "Version number was not present", None)
```
Did you mean to have 'request' in there?

Reply:
yes, that was typo.  I'll probably just remove these.  They look web-dev'y anyway

Reply:
I important consideration in this case is "what user actions might lead to this error and what can the user do about it?" It looks to me like this and the next error will only occur if the user creates an AraDefinition locally and tries to use it without registering it, since the act of registering will add a uid and a version number. We can therefore provide some sort of descriptive error message. It wouldn't be an ApiError--maybe a TypeError?

Reply:
`ValueError` seems reasonable for this

Reply:
seems to me like we need two types.  One for the definition, and another for a registered definition.  I think the current type conflates these two ideas

Reply:
`AraDefinition(name: str, description:str, ...)`

`RegisteredAraDefinition(ara_def: AraDefinition, definition_uid: UUID, version_uid: UUID...)`

Reply:
This method would ONLY process Registered definitions.

Reply:
While I agree that having that distinction would be useful, it might be worth noting that the UI doesn't distinguish between registered and unregistered across the board, so that would be introducing a new concept/workflow to users when dealing with AraDefinitions

Reply:
@anduill personally I think I would take that a step further and just only let the SDK create an `AraDefinition` from the backend's response (`register` would just look a lot like the constructor), buuuut that's not the pattern we've been using for the most part, and I don't think working that out is In Scope for this PR. For now raising a `ValueError` when the critical attributes are missing should be fine. Or if that bothers you, could just require the id/version as method parameters directly, and let the user sort out tracking them down.

Reply:
or, I can just throw an exception

Reply:
is that different from 'raising a `ValueError`'?